src/blockchain/src/blockchain.rs

```diff
@@ -65,6 +70,8 @@ impl Blockchain {
         chain.blocks.push(block);
         Self {
             trans_observers: Default::default(),
+            key_pair: keypair.clone(),
```

Comment:
Since this is a new field we should be consistent with the old one

```suggestion
            keypair: keypair.clone(),
```

Reply:
I don't have this code. On my end, they are consistent

```rust
        Self {
            trans_observers: Default::default(),
            key_pair: key_pair.clone(),
            local_id,
            block_observers: vec![],
            chain,
        }
```